src/ViewRenderer.php

```diff
@@ -539,14 +541,16 @@ private function extractControllerName(object $controller): string
             return $cache[$class];
         }
 
-        $regexp = '/((?<=controller\\\|controllers\\\)(?:[\w\\\]+)|(?:[a-z\d]+))controller$/iU';
-        if (!preg_match($regexp, $class, $m) || empty($m[1])) {
+        if (preg_match('/(?:.*controller\\\|.*controllers\\\)([\w\\\]+)controller$/iU', $class, $m) && !empty($m[1])) {
```

Comment:
```suggestion
        if (preg_match('/.*controllers?\\\([\w\\\]+)controller$/iU', $class, $m) && !empty($m[1])) {
```

Reply:
This broken tests. 

I tried this expression: `/(?:.*controller\\\|.*controllers\\\)([\w\\\]+)controller$/iU`. It's also broken tests.

Reply:
OK.